segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/AbstractInMemoryChunkStorageProvider.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.mocks;
+
+import io.pravega.segmentstore.storage.chunklayer.BaseChunkStorageProvider;
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Base class for InMemory mock implementations.
+ * Allows to simulate ChunkStorageProvider with different supported properties.
+ */
+abstract public class AbstractInMemoryChunkStorageProvider extends BaseChunkStorageProvider {
+    /**
+     * value to return when {@link AbstractInMemoryChunkStorageProvider#supportsTruncation()} is called.
+     */
+    @Getter
```

Comment:
I assume this is a mock that is never used in production code, correct? 

Even so, unless you move this class to the test package, I would suggest transforming these fields into abstract/virtual methods and force the derived classes to override them. In the state that they are right now, anyone can instantiate your class and change these fields arbitrarily.

Reply:
No I need them to be accessible to test code. I'll probably move this to test package.